src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -1135,36 +1135,51 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
         if extra_ssl_cert_provided and spec.generate_cert:
             raise OrchestratorError("Cannot provide ssl_certificate in combination with generate_cert")
 
+        # pick ip RGW should bind to
+        ip_to_bind_to = ''
+        if spec.only_bind_port_on_networks and spec.networks:
```

Comment:
I assume we keep `spec.only_bind_port_on_networks` to ensure backward compatibility (it's `false` by default) and make the implementation consistent with what already implemented for the other daemons.
I have a quick question though: wasn't sufficient to simply check `spec.networks` and execute the underlying code when `len(spec.network) > 0` ? In general, if you specify your `networks` in the spec it is assumed you do not want to bind on `*`, but honor the input defined in the spec.
I might have missed something, but I was wondering if `if spec.only_bind_port_on_networks and spec.networks:` might have been turned into a `if spec.networks` and remove (entirely) this parameter.

Reply:
The decision was just a combination of desire to retain old behavior when the flag isn't passed, and consistency between RGW and other services that support this flag. I already don't really like that RGW was the only service that could magically make use of networks (and even then, the fact that it had to be a new service and didn't work with spec modification + redeploy wasn't great) while nothing else supported it. I'm not removing that functionality since it was there and I don't want to risk breaking someone's deployment, but generally I'd like all the services to work the same, meaning I'd like users to have to set the additional `only_bind_port_on_networks` option to get this behavior.